.github/workflows/release-build.yaml

```diff
@@ -0,0 +1,84 @@
+name: Docker Build Publish
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-rc*'
+
+env:
+  GITHUB_PAT: "${{ secrets.KIVA_ROBOT_GITHUB_PAT }}"
+  AWS_REGION: "us-west-2"
+  AWS_ACCESS_KEY_ID: "${{ secrets.AWS_ACCESS_KEY_ID }}"
+  AWS_SECRET_ACCESS_KEY: "${{ secrets.AWS_SECRET_ACCESS_KEY }}"
+  SLACK_BOT_TOKEN: "${{ secrets.SLACK_TOKEN }}"
+  SLACK_CHANNEL: "eng-build-failures"
+
+jobs:
+  test-build:
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+      cancel-in-progress: true
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'
+    env:
+      AWS_ACCESS_KEY_ID: "${{ secrets.AWS_ACCESS_KEY_ID }}"
+      AWS_SECRET_ACCESS_KEY: "${{ secrets.AWS_SECRET_ACCESS_KEY }}"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - uses: actions/checkout@v3
+        with:
+          repository: kiva/github-actions
+          token: ${{ env.GITHUB_PAT }}
+          path: .github/
+      - uses: snow-actions/sparse-checkout@v1.2.0
+        with:
+          repository: kiva/marketplace-web-ui-ci
+          token: ${{ env.GITHUB_PAT }}
+          path: .docker
+          ref: main
+          patterns: |
+              resources/org/kiva/marketplaceWebUiCi/ui
+      - name: move files
+        run: |
+          mv .docker/resources/org/kiva/marketplaceWebUiCi/ui/Dockerfile .
+      - name: docker-build
+        uses: ./.github/actions/docker-build-push
+        if: github.event_name == 'pull_request'
+        with:
+          dockerfile_target: "release"
+          push: false
+  # Build and push the image to ECR
+  build:
+    runs-on: ubuntu-latest
+    # Run on merges to development or tag pushes (done by create-release workflow)
+    if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags')
```

Comment:
@mcstover if something other than GitHub is pushing tags, we may need to re-evaluate this condition.

Reply:
So if `v2.3.4` is pushed this will "see" it and build a new docker image and push that to Dockerhub?

Jenkins is still creating our semantic release tags on every merge to the `main` branch.

So, I see that It could create a failure if Jenkins has already built and pushed that image or the other way around too...

Maybe we could turn the push of the Jenkinsfile once we have both UI + CPS doing that via GH Actions. Just need to do both apps together...

Reply:
Yes I recommend preventing Jenkins from pushing tags as it will cause this workflow to run and likely cause issues with the `create-release` flow.

Reply:
We've got a PR up [here](https://github.com/kiva/marketplace-web-ui-ci/pull/89) to turn off automatic tagging on merges to main for UI + CPS!